src/snowflake/cli/_plugins/snowpark/snowpark_entity.py

```diff
@@ -1,16 +1,100 @@
-from typing import Generic, TypeVar
+import os
+from pathlib import Path
+from typing import Generic, Optional, TypeVar
 
+from snowflake.cli._plugins.nativeapp.artifacts import build_bundle
+from snowflake.cli._plugins.nativeapp.entities.application_package_child_interface import (
+    ApplicationPackageChildInterface,
+)
+from snowflake.cli._plugins.nativeapp.feature_flags import FeatureFlag
 from snowflake.cli._plugins.snowpark.snowpark_entity_model import (
     FunctionEntityModel,
     ProcedureEntityModel,
 )
 from snowflake.cli.api.entities.common import EntityBase
+from snowflake.cli.api.project.schemas.v1.native_app.path_mapping import PathMapping
 
 T = TypeVar("T")
 
 
-class SnowparkEntity(EntityBase[Generic[T]]):
-    pass
+# WARNING: The Function/Procedure entities are not implemented yet. The logic below is only for demonstrating the
+# required interfaces for composability (used by ApplicationPackageEntity behind a feature flag).
+class SnowparkEntity(EntityBase[Generic[T]], ApplicationPackageChildInterface):
```

Comment:
I'm using inheritance as an "implements interface", there is no real inheritance here. But I'm open for suggestions for a more elegant way -- can you give an example how it can be simplified?